src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -198,7 +195,7 @@ public void OnStartActivity(Activity activity, object payload)
                 activity.DisplayName = path;
 
                 // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md
-                if (this.httpSemanticConvention.HasFlag(HttpSemanticConvention.Old))
+                if (this.options.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old))
```

Comment:
You could extract these into two dedicated private `bool` variables and avoid checking if the `enum` has the flags repeatedly:
- `this.options.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old)`
- `this.options.HttpSemanticConvention.HasFlag(HttpSemanticConvention.New)`

Reply:
can do this as a follow up PR if that is ok.